cmds/core/wget/wget.go

```diff
@@ -45,29 +46,20 @@ func usage() {
 	os.Exit(2)
 }
 
-func run() (reterr error) {
+func run(arg string) (reterr error) {
 	log.SetPrefix("wget: ")
 
-	if flag.Parse(); flag.NArg() != 1 {
-		usage()
+	if arg == "" {
+		return errors.New("empty URL")
```

Comment:
If you create an error:
var ErrEmptyURL = errors.new("empty URL")
then you can test for that error in your tests, with errors.Is()
Also, no need to name the return parrameter, I think; all your returns provide the error result, so
func run(arg string) error {

Reply:
Updated, actually using errors.Is will not work, because HTTPClientCodeError has nil error when status is not 200.